src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -0,0 +1,808 @@
+                return self.selectedHeatmapProfile;
+            },
+            get clusterHeatmapButtonDisabled() {
+                return (self.sortMode.type === "heatmap" &&
+                    self.selectedHeatmapProfile === self.sortMode.clusteredHeatmapProfile);
+            },
+            get hideClusterHeatmapButton() {
+                return !self.molecularProfileIdToHeatmapTracks.get(self.selectedHeatmapProfile);
+            },
+            get heatmapGeneInputValue() {
+                return self.heatmapGeneInputValue;
+            },
+            get customDriverAnnotationBinaryMenuLabel() {
+                const label = AppConfig.oncoprintCustomDriverAnnotationBinaryMenuLabel;
+                if (!label || !self.props.customDriverMetadata.hasDriverAnnotations) {
+                    return undefined;
+                } else {
+                    return label;
+                }
+            },
+            get customDriverAnnotationTiersMenuLabel() {
+                const label = AppConfig.oncoprintCustomDriverAnnotationTiersMenuLabel;
+                if (!label || !self.props.customDriverMetadata.customDriverTiers
+                        || !self.props.customDriverMetadata.customDriverTiers.length) {
+                    return undefined;
+                } else {
+                    return label;
+                }
+            },
+            get customDriverAnnotationTiers() {
+                const tiers = self.props.customDriverMetadata.customDriverTiers;
+                if (!tiers || !tiers.length) {
+                    return undefined;
+                } else {
+                    return tiers;
+                }
+            },
+            get annotateCustomDriverBinary() {
+                return self.props.store.mutationAnnotationSettings.driverFilter;
+            },
+            get selectedCustomDriverAnnotationTiers() {
+                return self.props.store.mutationAnnotationSettings.driverTiers;
+            },
+            get columnMode() {
+                return self.columnMode;
+            },
+            get horzZoom() {
+                return self.horzZoom;
+            },
+        });
+    }
+
+    componentWillUnmount() {
+        this.putativeDriverSettingsReaction();
+        this.urlParamsReaction();
+    }
+
+    private buildControlsHandlers() {
+        const considerChangingDistinguishDrivers = action(()=>{
+            if (!this.props.store.mutationAnnotationSettings.oncoKb &&
+                !this.props.store.mutationAnnotationSettings.hotspots &&
+                !this.props.store.mutationAnnotationSettings.cbioportalCount &&
+                !this.props.store.mutationAnnotationSettings.cosmicCount) {
+                this.distinguishDrivers = false;
+                this.props.store.mutationAnnotationSettings.ignoreUnknown = false;
+            } else {
+                this.distinguishDrivers = true;
+            }
+        });
+
+        return observable({
```

Comment:
probably doesn't need to be observable unless these functions are going to be changed.

Reply:
note to self: doen